phoenix-core-client/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -800,6 +799,8 @@ public List<HRegionLocation> getTableRegions(byte[] tableName, byte[] startRowKe
                         && Bytes.compareTo(currentKey, endRowKey) >= 0) {
                         break;
                     }
+                    throwErrorIfQueryTimedOut(startRowKey, endRowKey, maxQueryEndTime,
```

Comment:
I think we can test this by setting a low value for a query timeout and creating a table with multiple regions either through salting or split points. PhoenixQueryTimeoutIT seems like a good candidate for this test.

Reply:
One of the tests in PhoenixQueryTimeoutIT is already covering this. However, the way `BaseResultIterators.setForTestingSetTimeoutToMaxToLetQueryPassHere` is being used is causing issues. We will have to clean it up or separate the test using it to different class. Given that it is static, it has potential to run into issues while running any other test in parallel.